### PR TITLE
Fix curl issue + change checks order

### PR DESCRIPTION
**Pull request by Annamikhlin**
Adding seperate function of curl and gpg packages installation for Ubuntu and Debian.
Removing "apt-transport-https" and "dirmngr" installation in case of Debian OS.
"user input", "check_os" and "os-release" checks order were changed.

### CODE REVIEW

File: server [hunk 1]
```diff
@@ -127,6 +127,17 @@ query_default_version() {
   DEFAULT_SCYLLA_VERSION=$(echo $DEFAULT_SCYLLA_VERSION_RAW | sed -e "s/.*version\":\"\(.*\)\".*/\1/g")
 }
 
+packages_update () {
+  OS_TYPE=$(cat /etc/os-release | awk -F '=' '/^NAME/{print $2}' | awk '{print $1}' | tr -d '"')
```
Comment (yaronkaikov) [inline server:131]: we are running  ` . /etc/os-release` in line 277

File: server [hunk 2]
```diff
@@ -127,6 +127,17 @@ query_default_version() {
...
+  if [ "$OS_TYPE" == "Ubuntu" ]; then
```
Comment (yaronkaikov) [inline server:132]: following the comment above you can change this line to 
```
if [ "$NAME" == "Ubuntu" ] ; then
```

File: server [hunk 3]
```diff
@@ -127,6 +127,17 @@ query_default_version() {
...
+    run_cmd apt update
+    run_cmd apt-get install $APT_FLAGS curl gnupg2
+  elif [ "$OS_TYPE" == "Debian" ]; then
```
Comment (yaronkaikov) [inline server:135]: Save as above

**Review by yaronkaikov — CHANGES_REQUESTED** [review]

File: server (hunk 1 above)
Comment (Annamikhlin) [reply server]: Right! Fixed. This line was removed and the lines of ". /etc/os-release" moved before calling the function

File: server (hunk 2 above)
Comment (Annamikhlin) [reply server]: Done. Fixed.

File: server (hunk 3 above)
Comment (Annamikhlin) [reply server]: Changed as above.

File: server [hunk 4]
```diff
@@ -127,6 +127,16 @@ query_default_version() {
   DEFAULT_SCYLLA_VERSION=$(echo $DEFAULT_SCYLLA_VERSION_RAW | sed -e "s/.*version\":\"\(.*\)\".*/\1/g")
 }
 
+packages_update() {
+  if [[ "$NAME" == *"Ubuntu"* ]]; then
+    run_cmd apt update
+    run_cmd apt-get install $APT_FLAGS curl gnupg2
+  elif [[ "$NAME" == *"Debian"* ]]; then
+    run_cmd apt update
+    run_cmd apt-get install $APT_FLAGS apt-transport-https curl gnupg2 dirmngr
```
[the remark this answers is not in the document]
Comment (Annamikhlin) [reply server:136]: Good question. "apt-transport-https" already removed from "Ubuntu" installation, maybe it is also not required for Debian? 
I will test it without "dirmngr" and "apt-transport-https".

File: server [hunk 5]
```diff
@@ -127,6 +127,16 @@ query_default_version() {
   DEFAULT_SCYLLA_VERSION=$(echo $DEFAULT_SCYLLA_VERSION_RAW | sed -e "s/.*version\":\"\(.*\)\".*/\1/g")
 }
 
+packages_update() {
+  if [[ "$NAME" == *"Ubuntu"* ]]; then
+    run_cmd apt update
+    run_cmd apt-get install $APT_FLAGS curl gnupg2
```
[the remark this answers is not in the document]
Comment (Annamikhlin) [reply server:133]: The commit message was updated.

File: server (hunk 4 above)
[the remark this answers is not in the document]
Comment (Annamikhlin) [reply server]: It works without "dirmngr" and "apt-transport-https" packages. 
I removed it from the code.

**Review by yaronkaikov — APPROVED** [review]
LGTM

**Comment by yaronkaikov** [comment]
@benipeled Please review

**Review by benipeled — APPROVED** [review]
LGTM, nice work with this change.

**Pull request merged**